packages/node/src/extensions/feature-flags/feature-flags.ts

```diff
@@ -582,21 +581,25 @@ class FeatureFlagsPoller {
   }
 
   async loadFeatureFlags(forceReload = false): Promise<void> {
-    // On first load, try to initialize from cache (if a cache provider is configured)
-    if (this.cacheProvider && !this.hasAttemptedCacheLoad) {
-      this.hasAttemptedCacheLoad = true
-      await this.loadFromCache('Loaded flags from cache')
+    if (this.loadedSuccessfullyOnce && !forceReload) {
+      return
     }
 
-    // If a fetch is already in progress, wait for it
-    if (this.loadingPromise) {
-      return this.loadingPromise
+    if (!this.loadingPromise) {
+      this.loadingPromise = (async () => {
+        if (this.cacheProvider) {
+          const cacheHit = await this.loadFromCache('Loaded flags from cache')
+          if (cacheHit) {
+            return
+          }
+        }
+        await this._loadFeatureFlags(true)
+      })().finally(() => {
+        this.loadingPromise = undefined
```

Comment:
I see `this.loadingPromise` is cleared in two places (here and line 773 in `_loadFeatureFlags`), which creates confusion.

More importantly, I notice the background poller at line 633 calls `_loadFeatureFlags()` directly:

```js
this.poller = setTimeout(() => this._loadFeatureFlags(), this.getPollingInterval())
```

This bypasses the loadingPromise guard in `loadFeatureFlags()`, which means:

1. The background poller can run concurrently with a manual call to `loadFeatureFlags()`
2. This could cause duplicate API fetches
3. The cleanup at line 773 might be trying to handle this, but it's unclear

Should the background poller also go through loadFeatureFlags() to use the promise guard? Or should it check if a load is already in progress?